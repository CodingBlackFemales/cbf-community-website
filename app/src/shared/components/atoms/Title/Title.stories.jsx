import React from 'react';

import { storiesOf } from '@storybook/react';

import Title from '.';

storiesOf('Atoms', module).add('Title', () => <Title text={'Blog'} />);
