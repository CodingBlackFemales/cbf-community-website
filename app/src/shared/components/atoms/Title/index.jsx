import React from 'react';
import styled from 'styled-components';

const Title = ({ text, className }) => {
  const classPrefix = 'title';
  return (
    <div className={(classPrefix, className)}>
      <h1 className={`${classPrefix}__text`}>{text}</h1>
    </div>
  );
};

const StyledTitle = styled(Title)`
  text-align: center;

  .title__text {
    text-transform: uppercase;
    font-weight: 300;
    margin-bottom: 0.2em;
  }
`;

export default StyledTitle;
