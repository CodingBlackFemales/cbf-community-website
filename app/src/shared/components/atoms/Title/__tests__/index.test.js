import React from 'react';
import { cleanup } from '@testing-library/react';

import Title from '..';
import {
  mountWithTheme,
  renderWithTheme,
} from '../../../../utils/test/testHelpers';

describe('<Title', () => {
  afterEach(cleanup);
  const titleComponent = <Title label={'A title'} />;

  it('matches the snapshot', () => {
    const { asFragment } = renderWithTheme(titleComponent);
    expect(asFragment()).toMatchSnapshot();
  });

  it('displays the header text and tagline', () => {
    const component = mountWithTheme(titleComponent);

    const titleText = component.find('.title__text');

    expect(titleText.exists()).toBe(true);
  });
});
