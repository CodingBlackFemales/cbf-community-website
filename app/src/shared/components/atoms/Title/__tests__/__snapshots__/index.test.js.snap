// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title matches the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  text-align: center;
}

.c0 .title__text {
  text-transform: uppercase;
  font-weight: 300;
  margin-bottom: 0.2em;
}

<div
    class="c0"
  >
    <h1
      class="title__text"
    />
  </div>
</DocumentFragment>
`;
